### DISCUSSION TITLE
How do I connect to a bus exposed at a certain IP and port?

### DISCUSSION THREAD

**Discussion by tobiasgehring** [python-sdbus/python-sdbus#78] [General]
I have a dbus daemon running a session bus that is accesssible at a certain IP address and port. How do I use the sdbus package to connect to it?

**Comment by igo95862** [comment]
Hello @tobiasgehring 

I believe that you might be able set the `DBUS_SESSION_BUS_ADDRESS` environment variable to something like `tcp:host=127.0.0.1`:

https://dbus.freedesktop.org/doc/dbus-specification.html#transports-tcp-sockets

However, this is not very secure. I would recommend using the sdbus over SSH instead:

https://python-sdbus.readthedocs.io/en/latest/common_api.html#sdbus.sd_bus_open_system_remote
